simulator/src/main.rs

```diff
@@ -87,6 +108,17 @@ struct Continuous {
     frame_time: u64,
 }
 
+#[derive(Clone, Parser)]
+struct Defined {
+    /// Path to the json settings file
+    #[clap(long)]
+    path: PathBuf,
```

Comment:
```suggestion
    file: PathBuf,
```

Also this would be better as an argument rather than an option.

Reply:
I'm not sure what you mean, as this is not an option. Do you mean it should be in the main Cli struct? It's only used for this mode however.

Reply:
This would commonly be regarded as an option (claps autogenerated help even does so) as it requires an identifier.

e.g. in `ls --sort size Downloads`, `--sort size` is an option and `Downloads` is an argument.

If you always need to specify something and it is the main thing you are acting on, then it is better to specify it as an argument.

Reply:
Ah, I see now